src/hammer-vlsi/hammer_vlsi/hammer_tool.py

```diff
@@ -931,35 +931,43 @@ def get_pin_assignments(self) -> List[PinAssignment]:
                 "Invalid pin_mode {mode}. Using none pin mode.".format(mode=pin_mode))
             return []
 
+        generate_mode = str(self.get_setting("vlsi.inputs.pin.generate_mode"))
+        if generate_mode not in ("full_auto", "semi_auto"):
```

Comment:
Hmm this is OK for now, but we may want to set a new standard for `generate_mode`s where the code that implements them defines the list of valid options. Maybe when/if we split `hammer_vlsi_impl.py` into multiple files (e.g. I could see a `pin_assignments.py` that contains a singleton `PinAssignmentModes` list).

Reply:
Do we actually envision more than these two? I feel that these two modes pretty much cover the major usecases we can think of for now.

Reply:
Yeah I can't really think of anything else.